CMakeLists.txt

```diff
@@ -186,6 +186,14 @@ if(S2GEOGRAPHY_BUILD_TESTS)
 
   FetchContent_MakeAvailable(nanoarrow)
 
+  FetchContent_Declare(
+    geoarrow
+    URL https://github.com/geoarrow/geoarrow-c/archive/refs/tags/geoarrow-c-python-0.1.3.zip
+    URL_HASH
+      SHA256=67a6781e1767ad666a0bcec5dcbd86356256a1dfd8aae9a54f4aee146e358ce6)
+
+  FetchContent_MakeAvailable(geoarrow)
```

Comment:
Another option could be to actually include the vendored geoarrow inside the s2geography sources and also install the headers as part of s2geography? (`s2geography/vendored/geoarrow.h`) And then we could use it that way instead of what I tried above? 

Or add some test utilities file in s2geography that exposes some of those geoarrow-c things through the standard package?

Reply:
> I think the fastest way to go about this for this PR would be something like `target_include_directories(geoarrow_test PRIVATE src/vendored)`

Added that. 

Now, looking at my conda env, it actually seems to install the geoarrow headers anyway even when it is for the test dependency, so not fully sure why I needed the `vendored/` part when first testing it.